src/os/ui/layer/defaultlayerui.js

```diff
@@ -433,6 +433,12 @@ os.ui.layer.DefaultLayerUICtrl.prototype.onRefreshChange = function() {
  */
 os.ui.layer.DefaultLayerUICtrl.prototype.reset = function(key) {
   var defaultValue = this.defaults[key];
+  var nodes = this.getLayerNodes();
+  var value = nodes[0].getLayer().getLayerOptions()['defaults'][key];
```

Comment:
This should be a bit more defensive instead of assuming everything in the call chain will be defined. The node list could be empty, `getLayer` and `getLayerOptions` are both allowed to return `null`, and `defaults` is currently only in layer options for layers created by a `LayerSyncDescriptor`. While it's admittedly unlikely a node would have a reset button and fall under these cases, it's worth verifying to avoid an exception.